firmware/src/core/utils/HTTPClientWrapper.h

```diff
@@ -0,0 +1,57 @@
+#ifndef HTTP_CLIENT_WRAPPER_H
+#define HTTP_CLIENT_WRAPPER_H
+
+#include <Arduino.h>
+#include <HTTPClient.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/semphr.h>
+#include <freertos/task.h>
+#include <freertos/queue.h>
+#include <functional>
+
+class HTTPClientWrapper {
+public:
+    using ResponseCallback = std::function<void(int httpCode, const String& response)>;
+    
+    static HTTPClientWrapper* getInstance();
+    bool makeRequest(const String& url, ResponseCallback callback);
+    bool addRequest(const String& url, ResponseCallback callback);
+    void processRequestQueue();  
+    void processResponseQueue();
+
+private:
+    // Add debug counters
+    static volatile uint32_t activeRequests;
+    static volatile uint32_t maxConcurrentRequests;
+
+    HTTPClientWrapper();
+    ~HTTPClientWrapper();
+    
+    static void httpTask(void* params);
+    
+    struct RequestParams {
+        String url;
+        ResponseCallback callback;
+    };
+
+    struct ResponseData {
+        int httpCode;
+        String response;
+        ResponseCallback callback;
+    };
+
+    static HTTPClientWrapper* instance;
```

Comment:
If the `HTTPClientWrapper` is already more or less a singleton, why have the individual fields below also be static?
Considering there is only one instance of this class, they can be members just fine?

Reply:
They could be member variables instead, I've always used static variables with a singleton as I never have to think about whether the data is shared or not.  In this case there's ever only one so it doesn't matter as much